uqcsbot/scripts/yt.py

```diff
@@ -0,0 +1,54 @@
+import os
+from uqcsbot import bot, Command
+import argparse
+
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+YOUTUBE_API_KEY = 'AIzaSyCtxVMs6So6x2WL5WkDV4rm01hzddCCGH4'
+YOUTUBE_API_SERVICE_NAME = 'youtube'
+YOUTUBE_API_VERSION = 'v3'
+YOUTUBE_VIDEO_URL = 'https://www.youtube.com/watch?v='
+
+
+@bot.on_command('yt')
+def handle_yt(command: Command):
+    '''
+    `!yt <QUERY>` - Returns the top search result based on the query string.
+    
+    handle_yt checks the usage of yt command and appropriately selects logic.
+    '''
+    # Makes sure the query is not empty.
+    if command.has_arg():
+        try:
+            cmd = command.arg.strip()
+            yt_normal(cmd, command.channel)
+        except HttpError as e:
+            # Googleapiclient should handle http errors
+            print('An HTTP error %d occurred:\n%s' % (e.resp.status, e.content))
+    else:
+        bot.post_message(command.channel, "You can't look for nothing. !yt <QUERY>")
+        
+def yt_normal(search_query: str, channel):
```

Comment:
can probably come up with more descriptive name like `get_top_search_result`

Reply:
and ideally this function should return the search result and have the parent function handle it (i.e. do the posting)

idc if you ceebs doing that, but would just be neater and separate functionality a bit nicer